solutions/usaco-712.mdx

```diff
@@ -160,7 +160,66 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
 **Time Complexity:** $\mathcal{O}(T^3)$, where $T$ represents the total number of characters in Farmer John's string.
 
 <LanguageSection>
-	
+
+<CPPSection>
+
+```cpp
+#include<bits/stdc++.h>
+
+const int TOTAL = 52;
+
+int main()
+{
+	freopen("circlecross.in", "r", stdin);
+	freopen("circlecross.out", "w", stdout);
+
+	std::string recordings;
+	std::cin >> recordings;
+
+	std::vector<int> first_found(26, -1);
+	std::set<std::pair<char, char>> crossings;
+
+	for (int i = 0; i < TOTAL; i++)
+	{
+		if (first_found[recordings[i] - 'A'] == -1)
+		{
+			// First occurence of this letter
+			first_found[recordings[i] - 'A'] = i;
+			continue;
+		}
+		else
+		{
+			// Make a list of all cows that have an entry/exit point
+			// between the entry and exit point of the current cow
+
+			std::set<char> appearances;
+
+			int start = first_found[recordings[i] - 'A'] + 1, end = i;
+			for (int j = start; j < end; j++)
+			{
+				if (appearances.count(recordings[j]))
```

Comment:
i recommend putting brackets around every statement

Reply:
> looks great! dont forget to add yourself to the author list :)

Hehe.. Thanks for telling

Reply:
> i recommend putting brackets around every statement
Ok I will put them up